java/arcs/core/policy/PolicyTranslation.kt

```diff
@@ -0,0 +1,37 @@
+package arcs.core.policy
+
+import arcs.core.analysis.RecipeGraph
+
+/**
+ * Translates the given [Policy] into dataflow analysis checks and claims on the given
+ * [RecipeGraph], returning a modified copy of the graph.
+ */
+fun applyPolicy(policy: Policy, graph: RecipeGraph): RecipeGraph {
+    checkEgressParticles(policy, graph)
+    // TODO(b/157605232): Modify the graph and attach checks and claims.
+    return graph
+}
+
+private fun checkEgressParticles(policy: Policy, graph: RecipeGraph) {
+    val invalidEgressParticles = graph.particleNodes
+        .map { it.particle.spec }
+        .filter { !it.isolated && it.name != policy.egressParticleName }
+    if (invalidEgressParticles.isNotEmpty()) {
+        throw PolicyViolation.InvalidEgressParticle(policy, invalidEgressParticles.map { it.name })
+    }
+}
+
+/** Indicates that a policy was violated by a recipe. */
+sealed class PolicyViolation(val policy: Policy, message: String) : Exception(
```

Comment:
Wouldn't it be better if we have these as data classes instead of exceptions? 

```
sealed class PolicyViolation(val policy: Policy) {
  data class InvalidEgressParticle(policy: Policy, val particleNames: List<String>): PolicyViolation (...) {...}
}
```

Reply:
Why is that better? I don't think you can throw non-throwable classes.

Reply:
I meant can we avoid using exceptions for indicating errors? Exceptions don't show up in the API specification and, therefore, make the code hard to read and maintain. 

Also, for better error reporting, we will want to collect more than one policy violation. e.g., more than one usage violation. Exceptions would make it somewhat harder to do that? 

I think by refactoring the code a little bit, we can avoid using exceptions. In addition to InvalidEgressParticleViolation, I can think of a `PolicyViolation` like `AllowedUsageViolation`. What other `PolicyViolation` did you have any mind?

Reply:
This is the only error that can happen during the policy translation process, as far as I know. Most other errors will be during the subsequent dataflow analysis pass. 

Can we leave it as an exception for now? It feels wrong to come up with a flexible error reporting system when for now we actually only have one error to report. Seems pretty easy (/mostly mechanical) to fix up later on.

Reply:
That is a fair point about not over designing when we have only one error to report. We should revisit when we start reporting the violations from DFA. I think we will need to convert a DFA check violation back into a corresponding policy violation, but we don't have to worry about it for the first version. 

Can you document the `applyPolicy` API about this exception being thrown? (I left a review comment [ca305c0#r450628892](https://github.com/PolymerLabs/arcs/pull/5647/commits/ca305c01c6638f4836e99144188bb1a447a0391b#r450628892) in https://github.com/PolymerLabs/arcs/pull/5647 that also has some implication on the `applyPolicy` API.)

(As a counter argument, if this is the only error, I would say that we should not even have the `PolicyViolation` class, but, let us leave it as is for now. :) As you say, this is mostly a mechanical change if we want to fix up later. )

Reply:
Added to the KDoc for applyPolicy. You're right about the PolicyViolation subclassing; I've been hoisted with my own petard!

I'll submit this as-is for now, and address other API issues in subsequent PRs.